app/models/category.rb

```diff
@@ -46,14 +47,19 @@ def full_name
 
   def post_count
     if has_children?
-      # Use `to_a` to call `Enumerable#sum` here.
-      # We can remove `to_a` after Rails 5. http://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-sum
-      children.to_a.sum {|child| child.posts.published.count }
+      ids = subtree.inject([]) do |memo, category|
```

Comment:
`AR::Relation#select` returns AR object. However it is too large object to be needed.

How about the following?
``` ruby
subtree.flat_map {|category| category.posts.published.pluck(:id) }.uniq.count
```